### PR TITLE
exclusive lock can aquires lock on expired request

### PR DESCRIPTION
**Pull request by PraveenKumarMahto**
Exclusive lock should check expiresAt in case of TTL is expired.

### CODE REVIEW

**Comment by michaelfinch** [comment]
Mind adding a test for this? Specifically to check that XLock succeeds when exclusive.acquired=true and exclusive.expiresAt <= currentTime?

**Comment by PraveenKumarMahto** [comment]
Added TestCase for the same.

**Comment by PraveenKumarMahto** [comment]
> Mind adding a test for this? Specifically to check that XLock succeeds when exclusive.acquired=true and exclusive.expiresAt <= currentTime?

Added TestCase for the same.

File: lock_test.go [hunk 1]
```diff
@@ -115,6 +115,24 @@ func TestLockExclusive(t *testing.T) {
 	if err != lock.ErrAlreadyLocked {
 		t.Errorf("err = %s, expected the lock to fail due to the resource already being locked", err)
 	}
+
+	// Create a lock with some resource name and some lockId
+	// that resource expires after some time
+	// then with same resource name and lockId can able to create lock.
+	err = client.XLock("resource4", "cccc", lock.LockDetails{TTL: 2})
```
Comment (michaelfinch) [inline lock_test.go]: To keep the tests as short as possible, can you make the TTL `1` here?

File: lock_test.go [hunk 2]
```diff
@@ -115,6 +115,24 @@ func TestLockExclusive(t *testing.T) {
...
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Waiting to expire the lock with resource name "resource4" and lockId "cccc".
+	time.Sleep(3 * time.Second)
```
Comment (michaelfinch) [inline lock_test.go]: Same as above, let's sleep for the shortest amount of time possible, like `1100 * time.Millisecond)`

File: lock_test.go [hunk 3]
```diff
@@ -115,6 +115,24 @@ func TestLockExclusive(t *testing.T) {
...
+
+	// Try to lock "reource4" with "cccc", which is already expired.
+	err = client.XLock("resource4", "cccc", lock.LockDetails{})
+	if err != nil {
+		t.Errorf("err = %s,fail to lock due to the resource already being locked", err)
```
Comment (michaelfinch) [inline lock_test.go]: nit: replace `err = %s,fail` with `err = %s, failed`

File: lock_test.go (hunk 1 above)
Comment (PraveenKumarMahto) [reply lock_test.go]: Yup, the testcase time should be as short as possible.

**Comment by michaelfinch** [comment]
Thank you!

File: lock_test.go (hunk 2 above)
Comment (PraveenKumarMahto) [reply lock_test.go]: yes,  we can keep it as small as possible.

File: lock_test.go (hunk 3 above)
Comment (PraveenKumarMahto) [reply lock_test.go]: made required changes.

**Pull request merged**